internal/core/data/reading_test.go

```diff
@@ -43,7 +43,7 @@ func TestGetAllReadingsOverLimit(t *testing.T) {
 
 	dbClient = newReadingsMockDB()
 
-	_, err := getAllReadings()
+	_, err := getAllReadings(mockLoggingClient)
```

Comment:
Commented here; applies elsewhere.

mockLoggingClient is defined in a different file.  It's usage complicates potential future refactoring of the current monolithic data package into a series of more cohesive subpackages.

In my opinion, all references to mockLoggingClient (in this file and elsewhere) should simply be replaced with its assigned value; i.e. logger.NewMockClient().  This would keep the tests "clean" by instantiating required dependencies within the scope of each test (instead of relying on an existing variable that may be tainted by its use in a test that ran prior).